jena-arq/src/main/java/org/apache/jena/sparql/serializer/QuerySerializer.java

```diff
@@ -142,7 +144,19 @@ public void visitAskResultForm(Query query)
         out.print("ASK") ;
         out.newline() ;
     }
-    
+
+    @Override
+    public void visitJsonResultForm(Query query) {
+        out.print("JSON {");
+        List<String> terms = new ArrayList<>();
+        for (Map.Entry<String, Object> entry : query.getJsonMapping().entrySet()) {
```

Comment:
Patch applied! Thanks Andy! TIL learned about that Prologue object.

Regarding the indentation, in `QuerySerializer.java`, there are other 2 occurrences of `incIndent`/`decIndent`, but instead of using `4`, they use the class constant

`static final int BLOCK_INDENT = 2 ;`

Should we use that for JSON as well?

Thanks!!!

Reply:
Sorry about the patch - I couldn't figure how to clone your repo because GH did not like the fact I already had a repo called "jena". Yes, a constant is better.  I thought that 4 looked nicer than 2 and it does not get indented further.  Just use `incIndent` twice!

Reply:
Is it ready to merge?  Any area to look at specially? I've used `qparse` on JSON queries and run a few test queries with all the cases I can think of.

Reply:
> Just use incIndent twice!

Haha, I like it. Done!

>Is it ready to merge? Any area to look at specially? I've used qparse on JSON queries and run a few test queries with all the cases I can think of.

I think so ! Only minor thing on my mind now are some expressions. We have not implemented support for queries with expressions like

`JSON { 'F' : sum(1+2)} WHERE { } `

Or other functions. Perhaps that could - if necessary - be implemented later?

Reply:
For now, Expressions can be done with `BIND` at the end of the `WHERE` pattern.

(`sum` is a group-aggregator -- whole different ball game!)

Reply:
Oh, got it! Nice! So no more comments from me :smiley: